File: OOP_Challenge/Challenge_01/src/main/java/test/local/model/impl/Shirts.java
package test.local.model.impl;

import test.local.model.Category;
import test.local.model.Product;

public class Shirts extends Product {
    public Shirts(long productId, String name, String description, int dollarPrice) {
        super(productId, name, description, dollarPrice, Category.SHIRTS);
    }
}

File: assignment1/Assignment1/shirts.cs
// Copyright (c) DEVCOP. All Rights Reserved.

namespace Assignment1.Services
{
    public class Shirts : Clothing
    {
        // specific properties for arms
        private readonly string category = "shirts";

        public Shirts(int productid, string name, string desc, int price, int arms)
        {
            this.ProductId = productid;
            this.Name = name;
            this.Category = this.category;
            this.Description = desc;
            this.DollarPrice = price;
            this.Arms = arms;
        }

        public Shirts(int id, string name, string desc, int price)
        {
            this.ProductId = id;
            this.Name = name;
            this.Category = this.category;
            this.Description = desc;
            this.DollarPrice = price;
            this.Arms = 1;
        }

        protected int Arms { get; set; }
    }
}
